### PR TITLE
Fix Persistent IP Indexing Edge Cases in NetworkManager

### PR DESCRIPTION
**Pull request by tholonious**

### CODE REVIEW

**Comment by tholonious** [comment]
* **#1386** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/1386?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @tholonious and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: lib/network_manager.go [hunk 1]
```diff
@@ -484,32 +477,6 @@ func (nm *NetworkManager) cleanupFailedOutboundConnection(connection Connection)
 	nm.cmgr.RemoveAttemptedOutboundAddrs(oc.address)
 }
 
-// ###########################
```
Comment (tholonious) [inline lib/network_manager.go:487]: This was dead code

File: lib/network_manager.go [hunk 2]
```diff
@@ -1039,30 +1010,43 @@ func (nm *NetworkManager) removeRemoteNodeFromIndexer(rn *RemoteNode) {
 	nm.GetNonValidatorOutboundIndex().Remove(rn.GetId())
 	nm.GetNonValidatorInboundIndex().Remove(rn.GetId())
 
-	// Try to evict the remote node from the validator index. If the remote node is not a validator, then there is nothing to do.
-	if rn.GetValidatorPublicKey() == nil {
```
[the remark this answers is not in the document]
Comment (lazynina) [reply lib/network_manager.go:1043]: Okay I see now. we still need to remove the node from the node indexer.

**Review by lazynina — APPROVED** [review]

File: lib/network_manager.go [hunk 3]
```diff
@@ -1039,30 +1010,43 @@ func (nm *NetworkManager) removeRemoteNodeFromIndexer(rn *RemoteNode) {
...
+	for ip, remoteNodeId := range allPersistentIpRemoteNodeIds {
+		if remoteNodeId == rn.GetId() {
+			nm.persistentIpToRemoteNodeIdsMap.Remove(ip)
+		}
 	}
 }
 
 func (nm *NetworkManager) Cleanup() {
+	// Clean up all RemoteNodes that have timed out or are left indexed due to some other
+	// unexpected failure.
```
[the remark this answers is not in the document]
Comment (tholonious) [reply lib/network_manager.go:1049]: Correct. The peer has disconnected, but for some reason it's till indexed / stored in the NetworkManager.

**Pull request merged**